### PR TITLE
AzureFunctionAppV2 Changes for Flex and fix race condition.

### PR DESCRIPTION
**Pull request by FinVamp1**
**Task name**: AzureFunctionAppv2

**Description**: Added Support for Flex Consumption and resolved race condition on Run From Package App Setting.

**Documentation changes required:** (Y) This adds support for a New Sku and has a new Parameter.

**Added unit tests:** (N) 

**Attached related issue:** (Y) https://github.com/microsoft/azure-pipelines-tasks/issues/19909 

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected

### CODE REVIEW

**Comment by FinVamp1** [comment]
azp /run

**Review by manolerazvan — CHANGES_REQUESTED** [review]
Please bump the task version

File: Tasks/AzureFunctionAppV2/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts [hunk 1]
```diff
@@ -62,7 +62,10 @@ export class BuiltInLinuxWebAppDeploymentProvider extends AzureRmWebAppDeploymen
             linuxFunctionAppSetting = linuxFunctionAppSetting + removeRunFromZipAppSetting;
         }
         var customApplicationSetting = ParameterParser.parse(linuxFunctionAppSetting);
-        isNewValueUpdated = await this.appServiceUtility.updateAndMonitorAppSettings(customApplicationSetting);
+        isNewValueUpdated = await this.appServiceUtilityExt.updateAndMonitorAppSettings(customApplicationSetting);
+
+        // add Output for isNewValueUpdated  
+        tl.debug(" Ouutput the value for App Settings isNewValueUpdated : " + isNewValueUpdated);
```
Comment (manolerazvan) [inline Tasks/AzureFunctionAppV2/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts:68]: Please correct the typo "Ouutput".

**Review by manolerazvan — APPROVED** [review]

File: Tasks/AzureFunctionAppV2/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts (hunk 1 above)
Comment (FinVamp1) [reply Tasks/AzureFunctionAppV2/deploymentProvider/BuiltInLinuxWebAppDeploymentProvider.ts:68]: Done

**Comment by e-karlsson** [comment]
Hi @FinVamp1 ! I stumbled across this PR after troubleshooting deployment of a Function app to new Flex Consumption. I am aware that Flex Consumption is in preview mode and the documentation says "Continuous deployment using Azure DevOps Tasks (AzureFunctionApp@2) is not supported". So what was this PR for?

When I run it, I see the following in the logs:
```
Got service connection details for Azure App Service:'<func app name>'
The Deployment Type option does not apply for Flex Consumption.
Successfully updated App Service configuration details
Successfully added release annotation to the Application Insight : <func app name>
App Service Application URL: https://<func app name>.azurewebsites.net/
Finishing: Azure Function App Deploy: <func app name>
```

But the Function app did not have any functions. I cannot find any documentation how to use it also, do you have it somewhere?

**Comment by FinVamp1** [comment]
Hello, @e-karlsson the way the task is designed is that the appType should be set to functionAppLinux and then the task will detect that the App is running on Flex Consumption and deploy to app using the Flex Consumption deployment endpoint. The aim of this PR was to add support for this while Flex Consumption is on Preview. Thanks for pointing out that the docs have not been updated and we will work on this soon. If you can share your app name or full deployment logs to me using finbarr AT microsoft.com and I will look at this for you.

**Pull request merged**